### PR TITLE
Add "Open Documentation" button to EditorProperty context menu

### LINKED ISSUE #3098: Add property context menu to Inspector
### Describe the project you are working on

A game that uses `AnimationTree`.

### Describe the problem or limitation you are having in your project

If you want to set parameter of some `AnimationTree` parameters, one have to write:

```gdscript
$AnimationTree.set("parameters/Seek/seek_position", 12.0)
```

The property path is quite long and no auto-complete available. But it's there in the property's tooltip :(

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a context menu for properties in the Inspector. Put options like "Copy Value", "Copy Path", and "Open in Help" in it.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

mock

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Possible, but not a few lines I think.

### Is there a reason why this should be core and not an add-on in the asset library?

It's a bit strange to download an plugin to solve the UX problem.

### PR DESCRIPTION
**Pull request by fire-forge**
Closes https://github.com/godotengine/godot-proposals/issues/3098

This PR adds an "Open Documentation" button to the EditorProperty context menu which takes you to that property's documentation entry.

- Adds icons to all buttons in the context menu.
- Capitalize "Pin Value".
- Remove transparency from "CopyNodePath" editor icon.

| Before | After |
---|---
| Before | After |

### CODE REVIEW

**Comment by fire-forge** [comment]
Rebased and resolved the merge conflicts.

**Comment by KoBeWi** [comment]
Needs rebase.

Also not sure if modifying EditorPropery is needed here; it could probably be simpler. But I can check properly after it's rebased.

**Comment by fire-forge** [comment]
> Also not sure if modifying EditorPropery is needed here; it could probably be simpler.

The property context menu is created and handled by EditorProperty, so I'm not sure if there's any other way to do it. But if you have any suggestions for a better way to do it I'm happy to make changes.

**Comment by KoBeWi** [comment]
No, the context menu is in EditorInspector. Instead of emitting a signal and letting the property call back with opening doc, maybe it's possible to get the doc based on property name, so the whole code is in the inspector.

**Comment by fire-forge** [comment]
Rebased. 

> No, the context menu is in EditorInspector.

@KoBeWi The context menu is in EditorProperty, not EditorInspector. The PopupMenu is created in `EditorProperty::_update_popup()` and there is no reference to the menu outside of EditorProperty.

https://github.com/godotengine/godot/blob/d26442e709f6361af9ac755ec9291bb43f2cd69b/editor/editor_inspector.cpp#L980-L987

> Instead of emitting a signal and letting the property call back with opening doc, maybe it's possible to get the doc based on property name, so the whole code is in the inspector.

There isn't really a good way to do this for two reasons:
* To open the doc page, it not only needs the property name but also the class name and whether it is a member or theme property. I could make it loop through all of the classes to find which one has a property/theme property that matches, but that would complicate things since we're already finding the class in the process of finding the doc for the tooltip.
* EditorProperty needs to know whether the doc exists or not so that it can know whether or not to add the "Open Documentation" button to the context menu.

File: editor/icons/CopyNodePath.svg [hunk 1]
```diff
@@ -1 +1 @@
-<svg height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><g fill="#e0e0e0" transform="translate(0 -1036.4)"><circle cx="3" cy="1048.4"/><path d="m2 1c-.55226.0001-.99994.4477-1 1v12c.0000552.5523.44774.9999 1 1h12c.55226-.0001.99994-.4477 1-1v-8l-5-5zm1 2h6v3c0 .554.44599 1 1 1h3v6h-10zm3 5-2 4h2l2-4zm4 0-2 4h2l2-4z" fill-opacity=".78431" transform="translate(0 1036.4)"/></g></svg>
+<svg height="16" viewBox="0 0 16 16" width="16" xmlns="http://www.w3.org/2000/svg"><g fill="#e0e0e0" transform="translate(0 -1036.4)"><circle cx="3" cy="1048.4"/><path d="m2 1c-.55226.0001-.99994.4477-1 1v12c.0000552.5523.44774.9999 1 1h12c.55226-.0001.99994-.4477 1-1v-8l-5-5zm1 2h6v3c0 .554.44599 1 1 1h3v6h-10zm3 5-2 4h2l2-4zm4 0-2 4h2l2-4z" transform="translate(0 1036.4)"/></g></svg>
```
Comment (KoBeWi) [inline editor/icons/CopyNodePath.svg:1]: Is this change intended? Doesn't look related.

File: editor/editor_inspector.cpp [hunk 2]
```diff
@@ -3488,6 +3503,10 @@ void EditorInspector::_property_selected(const String &p_path, int p_focusable)
 	emit_signal(SNAME("property_selected"), p_path);
 }
 
+void EditorInspector::_property_open_documentation(const String &p_doc_path) {
+	EditorNode::get_singleton()->open_documentation(p_doc_path);
```
Comment (KoBeWi) [inline editor/editor_inspector.cpp:3507]: Can be replaced by `ScriptEditor::get_singleton()->goto_help()`, see SceneTreeDock code.
This way you can avoid modifying EditorNode.

**Comment by KoBeWi** [comment]
Ok I checked the code more thoroughly now
and it's still overly complex xd

Sorry for the initial confusion, I mixed up EditorProperty with `editor_properties.cpp` (which is just more specialized code, but I thought the base EditorProperty is there too). Your solution is correct for the most part, but what can be simplified is opening the docs. You emit a signal from EditorProperty, which is then received in EditorInspector, which calls a method in EditorNode, which emits a new signal, which is connected in ScriptEditor to open doc. That's too many steps and too many unnecessary changes.

As I mentioned in one of the comments, check how scene tree dock handles this. It just calls a method directly in script editor to open a doc. You could do the same in EditorProperty, there is no reason to add any new signal at all (pinning does this, because its logic is more complex and tied to the inspector; opening a doc page is a simple operation).

**Comment by fire-forge** [comment]
Thanks for the feedback. I'll follow your suggestions to simplify the code.

File: editor/icons/CopyNodePath.svg (hunk 1 above)
Comment (fire-forge) [reply editor/icons/CopyNodePath.svg:1]: Yes that is intended. The CopyNodePath icon had some transparency which makes it look out of place with the other solid white icons.

File: editor/editor_inspector.cpp (hunk 2 above)
Comment (fire-forge) [reply editor/editor_inspector.cpp]: I changed it to use `ScriptEditor::get_singleton()->goto_help()` directly in EditorProperty. All of the changes in EditorNode and ScriptEditor are gone now.

**Review by KoBeWi — APPROVED** [review]
